api/saml/provider.py

```diff
@@ -367,4 +369,9 @@ def saml_callback(self, db, subject):
         return credential, patron, patron_data
 
 
+def validator_factory():
+    return SAMLSettingsValidator(SAMLMetadataParser())
+
+
 AuthenticationProvider = SAMLWebSSOAuthenticationProvider
+Validator = validator_factory
```

Comment:
Looking at this in context... since this is a function, I'd rather have the ValidatorFactory look for it under the name `validator`. `Validator` makes it look like a class.

Reply:
Changed logic of `PatronAuthenticationValidatorFactory` a bit:
1. First, it tries to instantiate a validator instance using a class name stored in `Validator`
2. Then, it tries to instantiate a new validator instance using a factory function `validator_factory`

I removed `Validator` from `api.saml.provider` so `validator_factory` will be used by `PatronAuthenticationValidatorFactory`